### DISCUSSION TITLE
pycaret silently exits without completeing pipeline

### DISCUSSION THREAD

**Discussion by redpoint13** [pycaret/pycaret#1181] [Q&A]
Pycaret is silently exiting without an error message or completing the script.  This time it failed after running StratifiedKFlold.  I've assured that scikit-learn==0.23.2.  Further, the silent fail does not always occur in the same place.  Often it will just quit after "Checking Exceptions".  

I've rebuilt my environment several times to try to fix the error with no luck.  I realize this is a cryptic message but I'm not sure what exactly to post to make finding the error productive.

The environment is:
aiohttp==3.7.4.post0
aiohttp-cors==0.7.0
aioredis==1.3.1
alembic==1.4.1
argon2-cffi==20.1.0
async-generator==1.10
async-timeout==3.0.1
attrs==20.3.0
awscli==1.19.53
azure-core==1.13.0
azure-storage-blob==12.8.0
backcall==0.2.0
bleach==3.3.0
blessings==1.7
blis==0.7.4
Boruta==0.3
botocore==1.20.53
cachetools==4.2.1
catalogue==1.0.0
catboost==0.25.1
certifi==2020.12.5
cffi==1.14.5
chardet==4.0.0
click==7.1.2
cliff==3.7.0
cloudpickle==1.6.0
cmaes==0.8.2
cmd2==1.5.0
colorama==0.4.3
colorful==0.5.4
colorlog==5.0.1
colorlover==0.3.0
confuse==1.4.0
cryptography==3.4.7
cufflinks==0.17.3
cycler==0.10.0
cymem==2.0.5
Cython==0.29.23
databricks-cli==0.14.3
decorator==4.4.2
defusedxml==0.7.1
docker==5.0.0
docutils==0.15.2
entrypoints==0.3
filelock==3.0.12
flake8==3.9.0
Flask==1.1.2
funcy==1.15
future==0.18.2
gensim==4.0.1
gitdb==4.0.7
GitPython==3.1.14
google-api-core==1.26.3
google-auth==1.29.0
google-cloud-core==1.6.0
google-cloud-storage==1.37.1
google-crc32c==1.1.2
google-resumable-media==1.2.0
googleapis-common-protos==1.53.0
gpustat==0.6.0
graphviz==0.16
greenlet==1.0.0
grpcio==1.37.0
gunicorn==20.1.0
hiredis==2.0.0
htmlmin==0.1.12
hyperopt==0.2.5
idna==2.10
ImageHash==4.2.0
imbalanced-learn==0.7.0
ipykernel==5.5.3
ipython==7.22.0
ipython-genutils==0.2.0
ipywidgets==7.6.3
isodate==0.6.0
itsdangerous==1.1.0
jedi==0.18.0
Jinja2==2.11.3
jmespath==0.10.0
joblib==1.0.1
jsonschema==3.2.0
jupyter-client==6.1.12
jupyter-core==4.7.1
jupyterlab-pygments==0.1.2
jupyterlab-widgets==1.0.0
kiwisolver==1.3.1
kmodes==0.11.0
lightgbm==3.2.1
llvmlite==0.36.0
Mako==1.1.4
MarkupSafe==1.1.1
matplotlib==3.4.1
mccabe==0.6.1
missingno==0.4.2
mistune==0.8.4
mlflow==1.15.0
mlxtend==0.18.0
msgpack==1.0.2
msrest==0.6.21
multidict==5.1.0
murmurhash==1.0.5
nbclient==0.5.3
nbconvert==6.0.7
nbformat==5.1.3
nest-asyncio==1.5.1
networkx==2.5.1
nltk==3.6.1
notebook==6.3.0
numba==0.53.1
numexpr==2.7.3
numpy==1.19.5
nvidia-ml-py3==7.352.0
oauthlib==3.1.0
opencensus==0.7.12
opencensus-context==0.1.2
optuna==2.7.0
packaging==20.9
pandas==1.2.4
pandas-profiling==2.11.0
pandocfilters==1.4.3
parso==0.8.2
patsy==0.5.1
pbr==5.5.1
pexpect==4.8.0
phik==0.11.2
pickleshare==0.7.5
Pillow==8.2.0
plac==1.1.3
plotly==4.14.3
preshed==3.0.5
prettytable==2.1.0
prometheus-client==0.10.1
prometheus-flask-exporter==0.18.1
prompt-toolkit==3.0.18
-e git+git@github.com:ddr-capital/ddr-prosper@0dc59d2d4ea7e7ecdfc26b0daff47751203dd4d1#egg=prosper2
protobuf==3.15.8
psutil==5.8.0
ptyprocess==0.7.0
py-spy==0.3.5
pyaml==20.4.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycaret==2.3.0
pycodestyle==2.7.0
pycparser==2.20
pyflakes==2.3.1
Pygments==2.8.1
pyLDAvis==3.2.2
pynndescent==0.5.2
pyod==0.8.7
pyparsing==2.4.7
pyperclip==1.8.2
pyrsistent==0.17.3
python-dateutil==2.8.1
python-editor==1.0.4
pytz==2021.1
PyWavelets==1.1.1
PyYAML==5.4.1
pyzmq==22.0.3
querystring-parser==1.2.4
ray==1.2.0
redis==3.5.3
regex==2021.4.4
requests==2.25.1
requests-oauthlib==1.3.0
retrying==1.3.3
rsa==4.7.2
s3transfer==0.3.7
scikit-learn==0.23.2
scikit-optimize==0.8.1
scikit-plot==0.3.7
scipy==1.5.3
seaborn==0.11.1
Send2Trash==1.5.0
shap==0.39.0
six==1.15.0
slicer==0.0.7
smart-open==5.0.0
smmap==4.0.0
spacy==2.3.5
SQLAlchemy==1.4.9
sqlparse==0.4.1
srsly==1.0.5
statsmodels==0.12.2
stevedore==3.3.0
tabulate==0.8.9
tangled-up-in-unicode==0.0.7
tensorboardX==2.2
terminado==0.9.4
testpath==0.4.4
textblob==0.15.3
thinc==7.4.5
threadpoolctl==2.1.0
tornado==6.1
tqdm==4.60.0
traitlets==5.0.5
tune-sklearn==0.2.1
typing-extensions==3.7.4.3
umap-learn==0.5.1
urllib3==1.26.4
visions==0.6.0
wasabi==0.8.2
wcwidth==0.2.5
webencodings==0.5.1
websocket-client==0.58.0
Werkzeug==1.0.1
widgetsnbextension==3.5.1
wordcloud==1.8.1
xgboost==1.4.0
yarl==1.6.3
yellowbrick==1.3.post1

**Comment by Yard1** [comment]
Is it possible for you to make a Colab notebook that can reproduce the issue? Also, it would be best if you made an issue instead of a question.

**Reply by redpoint13** [reply]
I've been able to find the errors with a decorator wrapper:
```

def catch_exceptions():
    def catch_exceptions_decorator(job_func):
        @functools.wraps(job_func)
        def wrapper(*args, **kwargs):
            try:
                return job_func(*args, **kwargs)
            except Exception:
                import traceback
                print(traceback.format_exc())
        return wrapper
    return catch_exceptions_decorator
```

The errors that have been coming up are from tabular.py lines 370 to 389.  I believe this is caused by the data sample not having all the possible values that have been defined.  Is there an issue with simply commenting out these checks?

```
        for i in ord_keys:
            if i not in data_cols:
                raise ValueError(
                    "Column name passed as a key in ordinal_features param doesnt exist."
                )

        for k in ord_keys:
            if data[k].nunique() != len(ordinal_features[k]):
                raise ValueError(
                    "Levels passed in ordinal_features param doesnt match with levels in data."
                )

        for i in ord_keys:
            value_in_keys = ordinal_features.get(i)
            value_in_data = list(data[i].unique().astype(str))
            for j in value_in_keys:
                if str(j) not in value_in_data:
                    raise ValueError(
                        f"Column name '{i}' doesn't contain any level named '{j}'."
                    )
```

**Comment by pycaret** [comment]
@redpoint13 Thanks for reporting. Will you please open an issue for this. Seems like a bug. Would like to investigate.
